src/core/utils/config.tsx

```diff
@@ -2,11 +2,30 @@ import { RootState, useSelector } from "../store";
 import { addConfigEntries } from "../config.slice";
 import withSuffix from "./withSuffix";
 
-export type UseConfigFunction = (key: string) => unknown;
+type ReturnValue<T> = T extends string ? string : T[];
+
+type UseConfigFunction = <T>(
+  key: string,
+  options?: {
+    transformer?: "jsonParse" | "stringToArray";
+  }
+) => ReturnValue<T>;
```

Comment:
I think you should just define the conditional return type directly since  it is not used anywhere else:
```suggestion
) =>  T extends string ? string : T[];
```